File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.js
function solution(n, arr1, arr2) {
    const first = binary(arr1,n)
    const second = binary(arr2,n)
    return first.map((i,idx1)=>{
        return [...i].map((j,idx2)=>{
            if([j,second[idx1][idx2]].some(item=>item==='1')) return '#'
            return " "
        }).join("")
    })
}

function binary(arr,n){
    const  returnArray = arr.map(item=>{
        let binary = item.toString(2)
        while(binary.length<n){
            binary = "0" + binary
        }
        return binary
    })
    return returnArray
}

File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    for i,j in zip(arr1,arr2):
        answer.append(bin(i|j)[2:].zfill(n).replace("1","#").replace("0"," "))
    return answer
    
#bin함수는 앞에 0을 채워주지 않기 때문에 0을 채워놓고 코드를 실행시켜야 한다.
#ex)2를 2진으로 바꾸면 ob010 x ob10 으로 출력됨